adb_graphics/figures/maps.py

```diff
@@ -520,30 +527,21 @@ def _wind_barbs(self, level):
 
         u, v = self.field.wind(level)
 
-        model = self.model_name
         tile = self.map.tile
 
         # Set the stride and size of the barbs to be plotted with a masked array.
-        if self.map.m.projection == 'lcc' and tile == 'full':
-            if model == 'HRRR-HI':
-                stride = 12
-                length = 4
+        if tile in ['full', 'AK', 'CONUS', 'NHemi']:
+            if u.shape[0] < u.shape[1]:
+                stride = int(round(u.shape[0] / 35))
             else:
-                stride = 30
-                length = 5
-        elif self.map.m.projection == 'rotpole' and tile == 'full':
-            if model == 'RRFS_NA_3km':
-                stride = 50
-                length = 4
-            else:
-                stride = 15
-                length = 4
-        elif self.map.model == 'global' and tile == 'full':
-            stride = 20
-            length = 4
+                stride = int(round(u.shape[1] / 35))
+            length = 5
         else:
             stride = TILE_DEFS[tile]["stride"]
             length = TILE_DEFS[tile]["length"]
+            if self.map.model == 'globalCONUS':
+                stride = int(round(stride / 2.5))
+                length = 5
```

Comment:
A new feature, a decent stride can be found if you divide the number of grid points on the shorter side by 35.  Subdomains are defined by lat,lon so the stride is set in the TILE_DEFS.  For the CONUS subdomains, further dividing by 2.5 works well.

Reply:
Done.